lib/deimos/active_record_consume/batch_consumption.rb

```diff
@@ -92,9 +101,9 @@ def deleted_query(records)
         @klass.unscoped.where(@klass.primary_key => keys)
       end
 
-      # @param _record [ActiveRecord::Base]
+      # @param _batch_record [BatchRecord]
       # @return [Boolean]
-      def should_consume?(_record)
+      def should_consume?(_batch_record)
```

Comment:
Why did this need to change? Did it actually change?

Reply:
It did change types from `ActiveRecord::Base` to `BatchRecord`. This is in order to validate `associations` that are not yet saved